knowledge_base/React-Best-Practices-And-Improvements.md

```diff
@@ -92,6 +92,8 @@ A component's name should be identical to the file name.
 
 Files should not house more than a single component.
 
+Per React best practices, we use class names rather than ids. Component class names should replicate their component name.
```

Comment:
This can be subject to change, as it is just my personal opinion, not exactly react best practices.

The issue here is that html ids are used mainly for a different paradigm of front end. One where when you run the javascript, you update the dom elements (through jquery).

The reason I did not go with this approach here is that react never has need for ids, as it uses the v-dom model. We could add ids to elements just for testing purposes, I just chose not to because I didn't want to clutter production code with cross cutting testing concerns. 

Anyways, the point I am trying to make here is that maybe we could get a second opinion on this or have a larger discussion here before codifying it into the docs?

Reply:
Yeah, I don't know if we should explicitly discourage using IDs. They can be really handy for automated testing as they have high specificity. A component html class can be duplicated in the DOM, but an ID cannot.

Reply:
Interesting! I was brought up, my first few years at Common, under a strict "we never use ids; always use class names" regime. I guess this has changed. (Granted, we wrote in Mithril then.)

@masvelio, you authored our React Best Practices doc, do you have opinions here?